### PR TITLE
feat(handlers): add getGatewayBot() & getApplicationInformation()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [x] Code changes have been tested against the Discord API, or there are no
  code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed,
    parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation,
  README, etc.

### CODE REVIEW

File: src/types/oauth.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { UserPayload } from "./guild.ts";
+import { TeamPayload } from "./teams.ts";
+
+export interface oauthApplication {
```
Comment (ayntee) [inline src/types/oauth.ts:4]:
```suggestion
export interface OAuthApplication {
```

File: src/api/handlers/oauth.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { RequestManager } from "../../rest/request_manager.ts";
+import { oauthApplication } from "../../types/oauth.ts";
+import { endpoints } from "../../util/constants.ts";
+
+/** Returns the bot's OAuth2 application object without `flags`. */
+export function getApplicationInformation() {
+  return RequestManager.get(endpoints.OAUTH2_APPLICATION) as Promise<
+    oauthApplication
+  >;
+}
```
Comment (ayntee) [inline src/api/handlers/oauth.ts:10]:
```suggestion
import { RequestManager } from "../../rest/request_manager.ts";
import { OAuthApplication } from "../../types/oauth.ts";
import { endpoints } from "../../util/constants.ts";

/** Returns the bot's OAuth2 application object without `flags`. */
export function getApplicationInformation() {
  return RequestManager.get(endpoints.OAUTH2_APPLICATION) as Promise<
    OAuthApplication
  >;
}
```

File: src/types/teams.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  team_id: string;
+  /** the avatar, discriminator, id, and username of the user */
+  user: Partial<UserPayload>;
+}
+
+/** https://discord.com/developers/docs/topics/teams#data-models-membership-state-enum */
+export enum MembershipState {
+  INVITED = 1,
+  ACCEPTED,
+}
```
Comment (ayntee) [inline src/types/teams.ts:31]: Irrelevant change.

File: src/api/handlers/gateway.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { RequestManager } from "../../rest/request_manager.ts";
-import { Application, DiscordBotGatewayData } from "../../types/mod.ts";
+import { DiscordBotGatewayData } from "../../types/mod.ts";
```
Comment (ayntee) [inline src/api/handlers/gateway.ts:2]: The name of the file has been changed, this is an irrelevant change as well.

**Review by ayntee — COMMENTED** [review]
Why is the `team.ts` file added with no content inside? Remove the file.

File: src/types/teams.ts (hunk 3 above)
Comment (itohatweb) [reply src/types/teams.ts:31]: Skillz wanted better types and so this is required.

File: src/api/handlers/gateway.ts (hunk 4 above)
Comment (itohatweb) [reply src/api/handlers/gateway.ts:2]: You mean we should keep `misc.ts`?

**Review by ayntee — APPROVED** [review]

**Pull request merged**